azurerm/internal/services/storage/blobs.go

```diff
@@ -346,3 +361,21 @@ func (sbu BlobUpload) blobPageUploadWorker(ctx context.Context, uploadCtx blobPa
 		uploadCtx.wg.Done()
 	}
 }
+
+func convertHexToBase64Encoding(str string) (string, error) {
+	data, err := hex.DecodeString(str)
+	if err != nil {
+		return "", fmt.Errorf("%s", err)
+	}
+
+	return base64.StdEncoding.EncodeToString(data), nil
+}
+
+func convertBase64ToHexEncoding(str string) (string, error) {
+	data, err := base64.StdEncoding.DecodeString(str)
+	if err != nil {
+		return "", fmt.Errorf("%s", err)
```

Comment:
could we wrap this error:

```suggestion
		return "", fmt.Errorf("converting %q from Base64 to Hex Encoding: %+v", str, err)
```